app/src/main/java/io/gnosis/safe/ui/safe/balances/SafeBalancesViewModel.kt

```diff
@@ -0,0 +1,46 @@
+package io.gnosis.safe.ui.safe.balances
+
+import io.gnosis.data.models.Safe
+import io.gnosis.safe.di.Repositories
+import io.gnosis.safe.ui.base.AppDispatchers
+import io.gnosis.safe.ui.base.BaseStateViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.collect
+
+class SafeBalancesViewModel @Inject constructor(
+    repositories: Repositories,
+    appDispatchers: AppDispatchers
+) : BaseStateViewModel<SafeBalancesState>(appDispatchers) {
+
+    private val safeRepository = repositories.safeRepository()
+
+    override fun initialState(): SafeBalancesState = SafeBalancesState.SafeLoading(null)
+
+    init {
+        safeLaunch {
+            safeRepository.activeSafeFlow().collect { safe ->
+                updateState {
+                    SafeBalancesState.ActiveSafe(
+                        safe, if (safe == null)
+                            ViewAction.NavigateTo(
+                                SafeBalancesFragmentDirections.actionSafeBalancesFragmentToNoSafeFragment()
```

Comment:
So for now I think it's fine, but I think it would be cool to have  viewHolder that acts as an empty state, so anywhere we have list we can use it.